### PR TITLE
Adding basic MKL-DNN code

### PR DESCRIPTION
**Pull request by mahmoud-abuzaina**
This PR includes basic code of MKL-DNN integration in the graph rewrite and the convolution op. It is the base for the upcoming MKL-DNN kernels.

### CODE REVIEW

File: tensorflow/core/kernels/mkl_tfconv_op.h [hunk 1]
```diff
@@ -57,6 +61,71 @@ class MklToTfOp : public OpKernel {
     VLOG(1) << "MKLToTFConversion complete successfully.";
   }
 
+#ifdef INTEL_MKL_DNN
+  static void ConvertMklToTf(OpKernel* op_kernel, OpKernelContext* context,
+                             string data_format_str, DataType op_data_type,
+                             bool has_avx512f, uint input_number) {
+    try {
```
Comment (andydavis1) [inline tensorflow/core/kernels/mkl_tfconv_op.h]: Google style-guide does not allow C++ exceptions.
https://google.github.io/styleguide/cppguide.html#Exceptions

**Review by andydavis1 — APPROVED** [review]
Looks good with one small change to fix up.

File: tensorflow/core/kernels/mkl_tfconv_op.h (hunk 1 above)
Comment (mahmoud-abuzaina) [reply tensorflow/core/kernels/mkl_tfconv_op.h]: @caisq can you start the test here? I think @andydavis1 is ok with the changes.

**Comment by gunan** [comment]
Jenkins, test this please.

File: tensorflow/core/kernels/mkl_tfconv_op.h (hunk 1 above)
Comment (gunan) [reply tensorflow/core/kernels/mkl_tfconv_op.h]: Note for posterity,
In opensource tensorflow, we enable exceptions only for MKL.
Rest of the code has exceptions disabled.

**Comment by mahmoud-abuzaina** [comment]
@gunan It seems the only failing test failed because of Timeout. I assume it is flaky.

**Comment by gunan** [comment]
Sorry for forgetting about this change, I have been OOO for a while.
retesting just in case.

**Review by gunan — APPROVED** [review]

**Comment by gunan** [comment]
@mahmoud-abuzaina Could you rebase your changes on master? I cant get a green test run again, and I am not sure if these are due to old failures, or some issues we may be introducing.

**Comment by mahmoud-abuzaina** [comment]
@gunan Done.

**Comment by gunan** [comment]
Looks like there are now some clang-format issues:

```
=== Sanity check step 11 of 11: do_clang_format_check (Clang Format Check: Check .h and .cc files with Google C++ style) ===

File tensorflow/core/graph/mkl_graph_util.h is not properly formatted with clang-format --style=google
File tensorflow/core/graph/mkl_layout_pass.cc is not properly formatted with clang-format --style=google
File tensorflow/core/graph/mkl_tfconversion_pass.cc is not properly formatted with clang-format --style=google
File tensorflow/core/kernels/mkl_conv_grad_filter_ops.cc is not properly formatted with clang-format --style=google
File tensorflow/core/kernels/mkl_conv_grad_input_ops.cc is not properly formatted with clang-format --style=google
File tensorflow/core/kernels/mkl_conv_ops.cc is not properly formatted with clang-format --style=google
File tensorflow/core/kernels/mkl_conv_ops.h is not properly formatted with clang-format --style=google
File tensorflow/core/kernels/mkl_tfconv_op.h is not properly formatted with clang-format --style=google
File tensorflow/core/util/mkl_util.h is not properly formatted with clang-format --style=google
File tensorflow/core/util/mkl_util_test.cc is not properly formatted with clang-format --style=google
```
Could you take a look?

**Comment by mahmoud-abuzaina** [comment]
@gunan it looks like the style checker has added more constraints. Looking at it.

**Comment by mahmoud-abuzaina** [comment]
@gunan I fixed clang-format style issues.

**Comment by gunan** [comment]
Jenkins, test this please

**Comment by mahmoud-abuzaina** [comment]
Not sure why the clang-format still reporting errors although I ran the clang-format tool on the files, and cpplint does not show errors.

**Comment by gunan** [comment]
Is it possible the difference comes from `--style=google` flag for clang-format?
@yifeif in case she has more insight.

**Comment by mahmoud-abuzaina** [comment]
I always use --style=google

**Comment by yifeif** [comment]
Looks like different clang-format version might format differently even with the same style option. #14115 to disable it for now.

**Comment by yifeif** [comment]
The check should be disabled now if we can rebase. Thanks!

**Comment by mahmoud-abuzaina** [comment]
Thanks @yifeif. I did a rebase.

**Comment by gunan** [comment]
Jenkins, test this please.

**Comment by gunan** [comment]
Jenkins, test this please.

**Comment by mahmoud-abuzaina** [comment]
@gunan I am still not able to verify if the failing tests are failing because of the MKL changes.

**Comment by gunan** [comment]
I think these are infra issues. I will get them investigated as soon as possible.

**Pull request merged**